packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -312,10 +312,16 @@ export abstract class FluidDataStoreContext
 			logger: this.logger,
 			namespace: "FluidDataStoreContext",
 			properties: {
-				all: tagCodeArtifacts({
-					fluidDataStoreId: this.id,
-					fullPackageName: this.pkg?.join("/"),
-				}),
+				all: {
+					...tagCodeArtifacts({
+						fluidDataStoreId: this.id,
+					}),
+					...tagCodeArtifacts({
```

Comment:
can't these be combined?

Reply:
That is what I was trying but couldn't make it work. `all` when creating logger accepts a getter as property but other places where we create an error or send logs don't.
So, the return type of `tagCodeArtifacts` has to be either a getter or not, it can't be both. I was trying to make the return type based on what the input type is but it did not work for me. So, I ended up doing this.

Reply:
i got something like this to work 
```
export const tagData = <
	T extends TelemetryDataTag,
	V extends Record<string, TelemetryEventPropertyTypeExt | (() => TelemetryEventPropertyTypeExt)>,
>(
	tag: T,
	values: V,
): {
	[P in keyof V]:
		| (V[P] extends () => TelemetryEventPropertyTypeExt
				? () => {
						value: ReturnType<V[P]>;
						tag: T;
				  }
				: {
						value: Exclude<V[P], undefined>;
						tag: T;
				  })
		| (V[P] extends undefined ? undefined : never);
} =>
	// eslint-disable-next-line @typescript-eslint/no-unsafe-return
	Object.entries(values)
		.filter((e) => e[1] !== undefined)
		// eslint-disable-next-line unicorn/no-array-reduce, unicorn/prefer-object-from-entries
		.reduce((pv, cv) => {
			const [key, value] = cv;
			pv[key] =
				// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
				typeof value === "function" ? () => ({ tag, value: value() }) : { tag, value };
			return pv;
		}, {}) as ReturnType<typeof tagData>;

export const tagCodeArtifacts = <
	T extends Record<string, TelemetryEventPropertyTypeExt | (() => TelemetryEventPropertyTypeExt)>,
>(
	values: T,
): ReturnType<typeof tagData> =>
	tagData<TelemetryDataTag.CodeArtifact, T>(TelemetryDataTag.CodeArtifact, values);
```

Reply:
my guess is what you were missing was the conditional typing: 
```
		| (V[P] extends () => TelemetryEventPropertyTypeExt
				? () => {
						value: ReturnType<V[P]>;
						tag: T;
				  }
				: {
						value: Exclude<V[P], undefined>;
						tag: T;
				  })
```

Reply:
Actually, @markfields and I spent a lot of time doing this and we did add this exact return type. What is different is that you removed this type after calling `Object.entries(values).filter` but we had that and that is what was not compiling for us :-).